src/objects/object.py

```diff
@@ -3,13 +3,32 @@
 
 
 class Object(Serializable):
-    __objects_count: int = 0
-    objects_dict = {}  # An empty object
+    '''
+    Abstract object existing in a `scene`. Represents part of scene's state,
+    in contrast to `Action` which represents only modifications of the state.
+    '''
 
-    def __init__(self, position) -> None:
+    def __init__(self, position: Point, scene) -> None:
         super().__init__(__class__.__name__)
-        self.properties.id: int = Object.__objects_count
+        self.properties.id: int = scene.objects_count
         self.properties.position: Point = position
+        self.scene = scene
+        self.walkable = False
 
-        Object.objects_dict[self.properties.id] = self
-        Object.__objects_count += 1
+        scene.add_object_to_map(self)
```

Comment:
i don't like passing scene to object constructor
it should be done outside object

Reply:
The `add_object_to_map` call could be easily moved outside `Object`. On the other hand the `Object` will still probably need referenece to scene. Other way would be to pass callbacks like `move` etc., but I rather have the reference.

Reply:
I won't change that now, if you still would like to see the changes we can make other task.